src/lib/server/routes/auth/deleteUser.ts

```diff
@@ -0,0 +1,42 @@
+import { auth } from '../../lucia';
+import type { BaseRequest, Context } from '../../helpers';
+import { setUserOnline, sendToContext } from '../../helpers';
+import { prisma } from '../../prisma';
+import type { User } from '@prisma/client';
+
+export const deleteUserCommand = 'auth/deleteUser';
+
+export interface DeleteUserRequest extends BaseRequest {
+  t: typeof deleteUserCommand;
+  data: { id: string };
+}
+
+export type DeleteUserResponse =
+  | { t: 'auth/deleteUserSuccess'; data: User }
+  | { t: 'auth/deleteUserFail' };
+
+export async function deleteUser(
+  context: Context,
+  request: DeleteUserRequest,
+): Promise<void> {
+  try {
+    await auth.invalidateSession(context.sessionId);
+    const { data } = request;
+    const user = await prisma.user.delete({ where: { id: data.id } });
```

Comment:
This is a possible breach - user can send any other user's id.
I recommend using `context.userId` instead here.